### PR TITLE
check command line mode before starting solution crawler and remote h…

### PR DESCRIPTION
**Pull request by heejaechang**
**Customer scenario**

random crash when user is building a solution/project using VS from command line

ex) devenv solution.sln /build

**Bugs this fixes:** 

https://github.com/dotnet/roslyn/issues/15490

**Workarounds, if any**

there is no workaround.

**Risk**

risk is very low since code only affects very specific mode by turning off features. 

**Performance impact**

it should actually improve perf on the mode a bit since it disables features that don't apply for the scenario (command line build)

**Is this a regression from a previous update?**
No

**Root cause analysis:**
Roslyn should check this special mode (command line build using VS) and behave differently than normal mode (VS running as normal)

How did we miss it?  What tests are we adding to guard against it in the future?
we probably need to add integration test that build a solution using vs

**How was the bug found?**
customer reported it.

### CODE REVIEW

**Comment by heejaechang** [comment]
I didn't make whole language service to not start since I wasn't sure end result of
such big change. that probably should be done by IDE team.

**Comment by heejaechang** [comment]
@jasonmalinowski @Pilchie can you take a look?

**Comment by jasonmalinowski** [comment]
The change itself seems fine and something we ultimately want, but I don't think we understand this crash yet. The code I saw had nothing to do with command line mode AFAIK and could crash in real scenarios too. Do we understand the causality here?

**Comment by heejaechang** [comment]
@jasonmalinowski that's true. let me fix that race as well.

**Comment by heejaechang** [comment]
@jasonmalinowski can you take a look? I removed the race as well.

File: src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs [hunk 1]
```diff
@@ -74,6 +73,9 @@ private partial class WorkCoordinator
                     _registration.Workspace.DocumentOpened += OnDocumentOpened;
                     _registration.Workspace.DocumentClosed += OnDocumentClosed;
                 }
+
+                // subscribe to option changed event
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs]: Can you comment why this fixes it and has to be here?  It's very not obvious...

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure are we running integration test by default now? @tmeschter is all integration test supposed to pass?

**Comment by tannergooding** [comment]
@heejaechang, yes they are running by default now (this is the new framework). Seems there is a bit of flakiness with deploying VSIXs as part of build still.

**Comment by heejaechang** [comment]
@tannergooding failed again.

**Comment by heejaechang** [comment]
@tmeschter  @tannergooding vs-integration test seems not ready to be on by default yet?

**Comment by heejaechang** [comment]
retest windows_release_vs-integration_prtest  please

**Comment by heejaechang** [comment]
windows_release_vs-integration_prtest is not stable. we can ignore this test.

**Pull request merged**